code/modules/mod/modules/_module.dm

```diff
@@ -258,10 +259,15 @@
 /obj/item/mod/module/proc/on_device_deletion(datum/source)
```

Comment:
you have to explain what the actual issue here is.
if the device is deleted, it gets set to null and then we delete the module
if the module is deleted, the devices signal is unregistered and then we delete the device.
how does this cause any sort of loop?

Reply:
In this case, it's because when the module would get deleted, the device could still be in the contents, and as such, unless you remove it from the contents, it'll get qdel'd a second time because of the way things work in `/atom/movable`.